### PR TITLE
Update github workflow

### PR DESCRIPTION
**Pull request by elfein727**
No clue if this works, but this is the idea I had. The `create-pull-request` API should handle uncommitted changes.

### CODE REVIEW

File: .github/workflows/rust-clippy.yml [hunk 1]
```diff
@@ -4,17 +4,17 @@
...
 name: rust-clippy analyze
 
 on:
   push:
-    branches: [ "master" ]
+    branches: ["master"]
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: [ "master" ]
+    branches: ["master"]
```
Comment (StellaSmith) [inline .github/workflows/rust-clippy.yml:17]: Doesn't make much sense to run this on pull requests

File: .github/workflows/rust-clippy.yml [hunk 2]
```diff
@@ -44,11 +44,19 @@ jobs:
...
-      - name: Upload analysis results to GitHub
-        uses: github/codeql-action/upload-sarif@v1
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        token: ${{ secrets.GITHUB_TOKEN }}
         with:
-          sarif_file: rust-clippy-results.sarif
-          wait-for-processing: true
+          # The new branch
+          branch: ${{ github.event.pull_request.title }}/clippy-fix
```
Comment (StellaSmith) [inline .github/workflows/rust-clippy.yml:56]: Instead of using the pull request title, maybe use `clippy-fix/<commit hash>`

**Review by StellaSmith — CHANGES_REQUESTED** [review]

File: .github/workflows/rust-clippy.yml (hunk 2 above)
Comment (elfein727) [reply .github/workflows/rust-clippy.yml:56]: Good idea.

**Comment by StellaSmith** [comment]
Also, i please sign you commits

**Comment by elfein727** [comment]
Will do.

File: .github/workflows/rust-clippy.yml (hunk 1 above)
Comment (elfein727) [reply .github/workflows/rust-clippy.yml]: You want that part of the old workflow removed?

**Pull request merged**